bot/cogs/verification.py

```diff
@@ -155,6 +162,26 @@ def bot_check(ctx: Context) -> bool:
         else:
             return True
 
+    @tasks.loop(hours=12)
+    async def periodic_ping(self) -> None:
+        """Post a recap message every week with an @everyone."""
+        messages = await self.bot.get_channel(Channels.verification).history(limit=5).flatten()  # check lasts messages
+        messages_content = [i.content for i in messages]
+        if PERIODIC_PING not in messages_content:  # if the bot did not posted yet
```

Comment:
It strikes me that there is some redundancy in this code in terms of loops. The loop inside the `else` can be used instead of the `messages_content` comprehension e.g.

```py
posted = False

async for message in messages:
    if message.content == PERIODIC_PING:
        posted = True
        ...

if not posted:
    ...send(PERIODIC_PING)
```

**Note that in this case the history iterator would not need to be flattened.**

I think an approach which would be better and still appropriate is to `break` after finding the message (in theory there should only ever be 0 or 1 of the periodic message in the history):

```py
async for message in messages:
    if message.content == PERIODIC_PING:
        ...
        break
else:
    ...send(PERIODIC_PING)
```